src/graph/UserExecutor.cpp

```diff
@@ -0,0 +1,321 @@
+}
+
+
+AlterUserExecutor::AlterUserExecutor(Sentence *sentence,
+                                   ExecutionContext *ectx) : Executor(ectx) {
+    sentence_ = static_cast<AlterUserSentence*>(sentence);
+}
+
+Status AlterUserExecutor::prepare() {
+    userItem_.set_account(sentence_->getAccount()->data());
+    auto opts = sentence_->getOpts();
+    for (auto &opt : opts) {
+        switch (opt->getOptType()) {
+            case WithUserOptItem::OptionType::LOCK:
+            {
+                userItem_.__isset.is_lock = true;
+                userItem_.set_is_lock(WithUserOptItem::asBool(opt->getOptVal()));
+                break;
+            }
+            case WithUserOptItem::OptionType::MAX_QUERIES_PER_HOUR:
+            {
+                userItem_.__isset.max_queries_per_hour = true;
+                userItem_.set_max_queries_per_hour(WithUserOptItem::asInt(opt->getOptVal()));
+                break;
+            }
+            case WithUserOptItem::OptionType::MAX_UPDATES_PER_HOUR:
+            {
+                userItem_.__isset.max_updates_per_hour = true;
+                userItem_.set_max_updates_per_hour(WithUserOptItem::asInt(opt->getOptVal()));
+                break;
+            }
+            case WithUserOptItem::OptionType::MAX_CONNECTIONS_PER_HOUR:
+            {
+                userItem_.__isset.max_connections_per_hour = true;
+                userItem_.set_max_connections_per_hour(WithUserOptItem::asInt(opt->getOptVal()));
+                break;
+            }
+            case WithUserOptItem::OptionType::MAX_USER_CONNECTIONS:
+            {
+                userItem_.__isset.max_user_connections = true;
+                userItem_.set_max_user_connections(WithUserOptItem::asInt(opt->getOptVal()));
+                break;
+            }
+        }
+    }
+    return Status::OK();
+}
+
+void AlterUserExecutor::execute() {
+    auto *mc = ectx()->getMetaClient();
+    auto future = mc->alterUser(userItem_);
+    auto *runner = ectx()->rctx()->runner();
+    auto cb = [this] (auto &&resp) {
+        if (!resp.ok()) {
+            DCHECK(onError_);
+            onError_(resp.status());
+            return;
+        }
+
+        DCHECK(onFinish_);
+        onFinish_();
+    };
+
+    auto error = [this] (auto &&e) {
+        LOG(ERROR) << "Exception: " << e.what();
+        onError_(Status::Error("Internal error"));
+    };
+
+    std::move(future).via(runner).thenValue(cb).thenError(error);
+}
+
+
+GrantExecutor::GrantExecutor(Sentence *sentence,
+                                   ExecutionContext *ectx) : Executor(ectx) {
+    sentence_ = static_cast<GrantSentence*>(sentence);
+}
+
+Status GrantExecutor::prepare() {
+    aclItem_ = sentence_->getAclItemClause();
+    type_ = toRole(aclItem_->getRoleType());
+    const auto& space = aclItem_->getSpaceName();
+    const auto& user = sentence_->getAccount();
+    auto spaceRet = ectx()->getMetaClient()->getSpaceIdByNameFromCache(space->data());
```

Comment:
Something depends the data can't call by `prepare`, here such as `getSpaceIdByNameFromCache` and  `getUserIdByNameFromCache `